light_prop/propagation/params.py

```diff
@@ -9,14 +9,24 @@ class ParamsValidationException(Exception):
 class PropagationParams:
     c = 299792458
 
-    def __init__(self, matrix_size, nu, wavelength, sigma, focal_length, distance, pixel):
+    def __init__(self, matrix_size, nu, wavelength, beam_diameter, focal_length, distance, pixel):
+        """
+        Validates and converts propagation params.
+        :param matrix_size: number of pixels on the side of square calculation matrix_size
+        :param nu: frequency in [GHz]
+        :param wavelength: wavelength of EM radiation in [mm]
+        :param beam_diameter: sigma parameter of Gaussian beam in [mm]
+        :param focal_length: focusing distance in [mm]
+        :param distance: propagation distance in [mm]
+        :param pixel: dimensions of the pixels used in calculations [mm]
+        """
         logging.info("Loading propagation params")
         self.matrix_size = matrix_size
         self.nu = nu
         if not wavelength:
             wavelength = PropagationParams.get_wavelength_from_nu(self.nu)
         self.wavelength = wavelength
```

Comment:
1. I'd rename `nu` to something more descriptive (same case as `sigma` elsewhere), maybe `wave_frequency` would be ok?
2. I don't know usages in details but `nu` and `wavelength` seem to be describing same piece of information yet they may not be in sych when user provide both
```python
PropagationParams(nu=321, wavelength=123) # nu does not correspond to wavelength
PropagationParams(nu=321) # ok
PropagationParams(wavelength = 123) # nu is None
```

Reply:
changed wavelength to optional param for now - it will be more explicit (Still mad developer can provide both^^ )